flask-server/main.py

```diff
@@ -233,12 +234,15 @@ def check_denylist(token):
 # add bookmark api
 @app.route('/apis/add-bookmark', methods=['POST'])
 def add_bookmark():
+    resp_token = decode_auth_token(request.json['Authorization'])
     entity = datastore.Entity(key=datastore_client.key('bookmark'))
     entity.update({
         'link': request.json['link'],
         'title': request.json['title'],
         'description': request.json['description'],
-        'deleted': False
+        'deleted': False,
+        'created': datetime.datetime.now(timezone.utc),
+        'creator': str(resp_token)
     })
     datastore_client.put(entity)
```

Comment:
Handle the case when decode_auth_token is not a valid resp_token, realistically won't happen because is already verified in the dashboard. Only case were this will come up is, if user deletes token in local storage after logging in and being verified. An user isn't re-checked and kicked out until they refresh the page.

Reply:
oo thanks!